pkg/apis/sources/v1alpha1/gitlabsource_types.go

```diff
@@ -48,8 +48,15 @@ type GitLabSourceSpec struct {
 	// to receive events from.
 	// Examples:
 	//   https://gitlab.com/gitlab-org/gitlab-foss
-	// +kubebuilder:validation:MinLength=1
-	ProjectURL string `json:"projectUrl"`
+	// +optional
+	ProjectURL string `json:"projectUrl,omitempty"`
+
+	// GroupURL is the url of the GitLab group for which we are interested
+	// to receive events from.
+	// Examples:
+	//   https://gitlab.com/gitlab-org/gitlab-foss
+	// +optional
+	GroupURL string `json:"groupUrl,omitempty"`
```

Comment:
I get your point. My initial idea was to have both properties and use `anyOf` in the CRD to prevent setting both.

Switching to `GitlabURL` would result in breaking changes what I wanted to prevent. If this is not a problem I'm fine with your approach.

Reply:
Yeah good point on the no breaking changes if possible - I didn’t know about `anyOf`, very cool!

Maybe before going one route it’s worth checking in with @matzew over whether a change to the CRD (even just introducing the new field) requires bumping the CRD version?

Reply:
Having thought about it more, let's go forwards with `anyOf` for now, and make the changes to the `v1beta1` CRD. If @matzew or Pierangelo (on vacation until the new year so I won't ping him here) think that these changes should be made to a `v1beta2` CRD it should be pretty straightforward to make that change if needed.

Thanks for all your work on this @arkadiusjonczek !!

Reply:
Okay great to hear that. So whats the next steps here?

Related to https://github.com/knative-extensions/eventing-gitlab/blob/main/config/300-gitlabsource.yaml#L78 the current CRD version is `v1alpha1`. So you want to update this from `v1alpha1` to `v1beta1` right?

Reply:
Ah if it's `v1alpha1` then we can leave it for now, I assumed it was `v1beta1`, my bad!

So next steps would be for you to update the CRD you linked above to have the `anyOf` validation, then I will take a second look at the PR.

Probably early in the new year another eventing approver will give you the second review needed for this to merge (I haven't done enough work in this repo to feel comfortable approving without a second look from someone else). 

Let me know if you have any questions, or if I can help in any way!